syntax = "proto3";

package bayes.bob;

message Association {
  // Name of the association.
  string name = 1;

  // Link to the association home page.
  string link = 2;

  // A set of filter IDs: do not show the association if one these filters
  // returns false for the project.
  // See the list of available filters in scoring.py.
  repeated string filters = 3;
}

message Associations {
  // A sorted list of associations. The ones that Bayes thinks are the best
  // suited are first.
  repeated Association associations = 1;
}

message VolunteeringMission {
  // Name of the association that requested the mission.
  string association_name = 1;

  // Short title for this mission.
  string title = 2;

  // Longer description for the mission, may use markdown.
  string description = 3;

  // Link to the mission page with full info and way to apply.
  string link = 4;

  // Whether the mission can be done from anywhere.
  bool is_available_everywhere = 5;
}

message VolunteeringMissions {
  // List of missions sorted by relevance for the user, or project.
  repeated VolunteeringMission missions = 1;
}

// A simple link to an external page.
message SimpleLink {
  // Path part of the URL of the page.
  string path = 1;
}
